### ISSUE TITLE
[Feature Request]: Request for Upgrade to Support Gradient Accumulation Steps in Latest SentenceTransformers Version

### ISSUE THREAD

**Issue by NoahAi25** [run-llama/llama_index#14989]
### Feature Description

Hello,

I would like to train my embedding model, but due to memory limitations, I cannot use a large batch size. I have read that gradient accumulation steps can help with this issue. This training argument is available in version 3.0.1 of sentenceTransformers, but the llama-index package does not support the latest version. Is it possible to get an upgrade or any suggestions for using llama-index with this feature?

Thank you.

### Reason

Llama does not support the latest version of SentenceTransformer and the new training API.
